idx/memory/memory.go

```diff
@@ -1660,7 +1701,21 @@ func (m *UnpartitionedMemoryIdx) Prune(now time.Time) ([]idx.Archive, error) {
 	}
 	m.RUnlock()
 
-	var pruned []idx.Archive
+	var pruned []*idx.ArchiveInterned
+
+	// expandPruned takes a list of interned archives and appends them to the
+	// pruned slice after calling NewSafeArchive() for each of them
+	expandPruned := func(defs []*idx.ArchiveInterned) {
+		prunedOld := pruned
+		pruned = make([]*idx.ArchiveInterned, len(prunedOld)+len(defs))
+		for i := 0; i < len(prunedOld); i++ {
+			pruned[i] = prunedOld[i]
+		}
+		for i := range defs {
+			pruned[len(prunedOld)+i] = idx.NewSafeArchive(defs[i])
```

Comment:
Will the call to `len(prunedOld)` be optimized away? Otherwise maybe we should just get the length once outside of the loop.

Reply:
do you mean optimized as in "the function call gets omitted"? i don't know how `len()` is implemented in the run time, but since this is part of the language and an extremely simple function (just read the .Len property of the slice header) i assume that's done in some optimized way